core/regions/src/main/java/software/amazon/awssdk/regions/internal/DefaultServiceEndpointKey.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.regions.internal;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.regions.EndpointTag;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.regions.ServiceEndpointKey;
+import software.amazon.awssdk.utils.ToString;
+import software.amazon.awssdk.utils.Validate;
+
+@SdkInternalApi
+public class DefaultServiceEndpointKey implements ServiceEndpointKey {
+    private final Region region;
+    private final Set<EndpointTag> tags;
+
+    private DefaultServiceEndpointKey(Builder builder) {
+        this.region = Validate.paramNotNull(builder.region, "region");
+        this.tags = new LinkedHashSet<>(Validate.paramNotNull(builder.tags, "tags"));
+        Validate.noNullElements(builder.tags, "tags must not contain null.");
+    }
+
+    @Override
+    public Region region() {
+        return region;
+    }
+
+    @Override
+    public Set<EndpointTag> tags() {
+        return Collections.unmodifiableSet(tags);
```

Comment:
Same here, might be better to wrap it in the ctor

Reply:
Done